presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java

```diff
@@ -192,21 +208,93 @@ public static PrestoSparkQueryRunner createSpilledHivePrestoSparkQueryRunner(Ite
         properties.put("experimental.spiller-spill-path", Paths.get(System.getProperty("java.io.tmpdir"), "presto", "spills").toString());
         properties.put("experimental.spiller-threads", Integer.toString(NODE_COUNT * TASK_CONCURRENCY));
         properties.putAll(additionalConfigProperties);
-        return createHivePrestoSparkQueryRunner(tables, properties);
+        return createHivePrestoSparkQueryRunner(tables, properties, hiveProperties);
     }
 
-    public static PrestoSparkQueryRunner createHivePrestoSparkQueryRunner(Iterable<TpchTable<?>> tables, Map<String, String> additionalConfigProperties)
+    public static PrestoSparkQueryRunner createHivePrestoSparkQueryRunner(Iterable<TpchTable<?>> tables, Map<String, String> additionalConfigProperties, Map<String, String> hiveProperties)
     {
-        PrestoSparkQueryRunner queryRunner = new PrestoSparkQueryRunner("hive", additionalConfigProperties);
+        PrestoSparkQueryRunner queryRunner = new PrestoSparkQueryRunner("hive", additionalConfigProperties, hiveProperties);
         ExtendedHiveMetastore metastore = queryRunner.getMetastore();
         if (!metastore.getDatabase(METASTORE_CONTEXT, "tpch").isPresent()) {
             metastore.createDatabase(METASTORE_CONTEXT, createDatabaseMetastoreObject("tpch"));
-            copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), tables);
-            copyTpchTablesBucketed(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), tables);
+            boolean isDWRFformat = hiveProperties.getOrDefault("hive.storage-format", "").equals("DWRF");
+            if (isDWRFformat) {
+                // DWRF doesn't support date type. Convert date columns to varchar for lineitem and orders.
+                createLineitem(queryRunner);
```

Comment:
Is this a copy-paste from somewhere else? Would it be possible to refactor the code to avoid copy-paste?

Reply:
It's from the `HiveExternalWorkerQueryRunner`, but since HiveExternalWorkerQueryRunner is a standalone class and in presto-native-execution package, probably not worth to do a large refactoring to share this small amount of code IMO